### PR TITLE
Initial Windows support

### PR DESCRIPTION
**Pull request by OllieJC**
This adds support for building, testing and running on Windows. Tested on Windows 11 (Windows Terminal with [nvm-windows](https://github.com/coreybutler/nvm-windows)) and Ubuntu, macOS test TBD.

The main change is replacing the requires for `config/paths.json` with `lib/paths.js` that returns a _path.normalize_'d version of the paths object. Replaced instances of `blah + '/'` or `blah + blah2` with `path.join(blah, blah2)` as `path` handles the OS support, meaning the correct joins are returned.

Tested:
- `npm install`
- `npm run lint`
- `npm run test`
- `npm run build`
- `npm start`

Related issues:
- https://github.com/alphagov/govuk-design-system/issues/694
- https://github.com/alphagov/govuk-design-system/issues/598

~Keeping in draft until tested on macOS, but making as a PR for others to test!~ Edit: @lfdebrux has tested successfully on macOS.

### CODE REVIEW

**Comment by lfdebrux** [comment]
Hi @OllieJC, thanks for the PR!

I've tested this on my GDS managed macOS laptop and it works for me!

I'm a little unsure about the change from `require('../config/paths.json')` to `require('../lib/paths.js')` though, as it's an idiom we use in a couple of other repos and if we change it in one we might want to change it everywhere to avoid confusion... I will talk to the other devs though and see what they think.

File: lib/metalsmith-title-checker.js
Comment (OllieJC) [reply lib/metalsmith-title-checker.js:40]: My main thinking was it only filters out matches, so seemed easier to add another even if it's redundant for Linux users and vice versa, instead of requiring the paths library and then using `path.normalize` - I'd be happy to change though

**Comment by OllieJC** [comment]
> Hi @OllieJC, thanks for the PR!
> 
> I've tested this on my GDS managed macOS laptop and it works for me!
> 
> I'm a little unsure about the change from `require('../config/paths.json')` to `require('../lib/paths.js')` though, as it's an idiom we use in a couple of other repos and if we change it in one we might want to change it everywhere to avoid confusion... I will talk to the other devs and see what they think.

Great! Any thoughts on the lib change? Benefits are that if copying an existing file's code then it'll include the normalised paths; without something similar then there would need to be a lot more `path.normalize` or similar added where paths are used.

**Comment by lfdebrux** [comment]
> > Hi @OllieJC, thanks for the PR!
> > I've tested this on my GDS managed macOS laptop and it works for me!
> > I'm a little unsure about the change from `require('../config/paths.json')` to `require('../lib/paths.js')` though, as it's an idiom we use in a couple of other repos and if we change it in one we might want to change it everywhere to avoid confusion... I will talk to the other devs and see what they think.
> 
> Great! Any thoughts on the lib change? Benefits are that if copying an existing file's code then it'll include the normalised paths; without something similar then there would need to be a lot more `path.normalize` or similar added where paths are used.

Sorry, been ill, hadn't got around to it, just chatted with the other devs now, we're mulling it over!

**Comment by vanitabarrett** [comment]
Sorry for the slow progress on this @OllieJC. The team have been focused on an upcoming breaking release and we haven't really had the headspace to look at this PR properly yet and discuss it. We're hoping to get the release out this week, and we'll take a look at this PR once that's out of the way 🙂

**Comment by vanitabarrett** [comment]
The change does make sense to me - I think it's a neater solution than normalising the path each time it's used. Whether we make the same change in our other repos which use `paths.json` (which from my looking might just be govuk-frontend) probably depends on whether we think those repos should 'work' on Windows (i.e: do we expect people to be using them with Windows machines). 

As discussed in dev catch-up, there might be some overhead to supporting Windows more than just this PR, so I'm not sure if we'd want to extend this change to `govuk-frontend` just yet. Perhaps we can make the change here to start with, and see how it goes? @lfdebrux

**Review by vanitabarrett — APPROVED** [review]
Thanks for raising this PR @OllieJC - sorry it's taken us a while to review it! This just needs a second review from someone else in the Design System team before it's ready to be merged.

**Comment by OllieJC** [comment]
Amazing, thanks @vanitabarrett and @lfdebrux. I've marked it as ready to review as it was still in draft state.

**Pull request merged**
